client/components/language-picker/modal.jsx

```diff
@@ -3,22 +3,121 @@
 /**
  * External dependencies
  */
-
 import PropTypes from 'prop-types';
 import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
 import classNames from 'classnames';
 import { localize } from 'i18n-calypso';
-import { find, includes, map, noop, partial, startsWith, isEmpty } from 'lodash';
+import {
+	find,
+	includes,
+	isEmpty,
+	map,
+	noop,
+	partial,
+	some,
+	startsWith,
+} from 'lodash';
 
 /**
  * Internal dependencies
  */
+import { getGeoCountryShort } from 'state/geo/selectors';
+import QueryGeo from 'components/data/query-geo';
 import Dialog from 'components/dialog';
 import SectionNav from 'components/section-nav';
 import SectionNavTabs from 'components/section-nav/tabs';
 import SectionNavTabItem from 'components/section-nav/item';
 import Search from 'components/search';
 
+/*
+ * List of territories that languages are grouped into (displayed as tabs in the UI).
+ * Source of this data is the CLDR database, namely the `node-cldr` NPM package.
+ *
+ * `subTerritories` properties map the territory names into numerical geographic region
+ * codes according to the UN M49 (https://unstats.un.org/unsd/methodology/m49/) standard
+ * that is used in the CLDR database data.
+ *
+ * To move languages between territories or to include a language in multiple territories
+ * change the 'territories' arrays in language data in 'config/_shared.json'.
+ *
+ * Some languages, e.g., Esperanto are not included in any territory -- these are
+ * discoverable only with search.
+ *
+ * `countries` is a list of countries in each region. It's used to map country code
+ * retrieved from geolocation service into the name of the tab selected by default.
+ */
+const TERRITORIES = [
+	{
+		id: 'africa-middle-east',
+		name: translate => translate( 'Africa and Middle East' ),
+		subTerritories: [ '145', '002' ],
+		countries: [
+			'AE', 'AM', 'AZ', 'BH', 'CY', 'GE', 'IL', 'IQ', 'JO', 'KW',
+			'LB', 'OM', 'PS', 'QA', 'SA', 'SY', 'TR', 'YE', 'DZ', 'EG',
+			'EH', 'LY', 'MA', 'SD', 'TN', 'EA', 'IC', 'BF', 'BJ', 'CI',
+			'CV', 'GH', 'GM', 'GN', 'GW', 'LR', 'ML', 'MR', 'NE', 'NG',
+			'SH', 'SL', 'SN', 'TG', 'AO', 'CD', 'CF', 'CG', 'CM', 'GA',
+			'GQ', 'ST', 'TD', 'BI', 'DJ', 'ER', 'ET', 'KE', 'KM', 'MG',
+			'MU', 'MW', 'MZ', 'RE', 'RW', 'SC', 'SO', 'SS', 'TZ', 'UG',
+			'YT', 'ZM', 'ZW', 'BW', 'LS', 'NA', 'SZ', 'ZA'
+		],
+	},
+	{
+		id: 'americas',
+		name: translate => translate( 'Americas' ),
+		subTerritories: [ '019' ],
+		countries: [
+			'BM', 'CA', 'GL', 'PM', 'US', 'BZ', 'CR', 'GT', 'HN', 'MX',
+			'NI', 'PA', 'SV', 'AG', 'AI', 'AW', 'BB', 'BL', 'BQ', 'BS',
+			'CU', 'CW', 'DM', 'DO', 'GD', 'GP', 'HT', 'JM', 'KN', 'KY',
+			'LC', 'MF', 'MQ', 'MS', 'PR', 'SX', 'TC', 'TT', 'VC', 'VG',
+			'VI', 'AR', 'BO', 'BR', 'CL', 'CO', 'EC', 'FK', 'GF', 'GY',
+			'PE', 'PY', 'SR', 'UY', 'VE'
+		],
+	},
+	{
+		id: 'asia-pacific',
+		name: translate => translate( 'Asia-Pacific' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **95** times:
`translate( 'Pacific' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).